### PR TITLE
feat: add and release helm chart

### PR DESCRIPTION
**Pull request by HamzaZo**
+ Add helm chart 
+ Publish helm chart to Github Pages

### CODE REVIEW

File: .github/workflows/publish-helm.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Helm
+        uses: azure/setup-helm@v1
+        with:
+          version: v3.5.2
+
+      - name: Publish Helm charts
+        uses: stefanprodan/helm-gh-pages@master
```
Comment (aslafy-z) [inline .github/workflows/publish-helm.yml]: I would recommend using the official helm publisher action: https://github.com/helm/chart-releaser-action

File: .github/workflows/publish-helm.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  publish-release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Helm
+        uses: azure/setup-helm@v1
+        with:
+          version: v3.5.2
```
Comment (aslafy-z) [inline .github/workflows/publish-helm.yml]: You don't have to pin the version without a specific use case, `setup-helm` will automatically take the latest stable version.

File: .github/workflows/publish-helm.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: "Publish: helm chart"
```
Comment (aslafy-z) [inline .github/workflows/publish-helm.yml:1]: This workflow does not need to run on each main branch deployment, only on release (tags).

**Comment by aslafy-z** [comment]
Some tools (for eg. `yq`) could be used to bump the chart [`version`](https://github.com/Carrefour-Group/azure-resourcemanager-exporter/pull/2/files#diff-0ba27df9237fcad4b64a02dfbe80b9d351e42db61ad807f81b1f944916c93ff6R4), [`appVersion`](https://github.com/Carrefour-Group/azure-resourcemanager-exporter/pull/2/files#diff-0ba27df9237fcad4b64a02dfbe80b9d351e42db61ad807f81b1f944916c93ff6R5) and [`image.tag`](https://github.com/Carrefour-Group/azure-resourcemanager-exporter/pull/2/files#diff-4fa26370e1cc930478629a13d25955fd38be052de743241b24f55e6f97f0a82cR17) at release time. This make the release process easier.

File: .github/workflows/publish-helm.yml (hunk 2 above)
Comment (HamzaZo) [reply .github/workflows/publish-helm.yml]: I used what has been done in `lint-helm.yml`

Comment (HamzaZo) [reply .github/workflows/publish-helm.yml]: done

File: .github/workflows/publish-helm.yml (hunk 3 above)
Comment (HamzaZo) [reply .github/workflows/publish-helm.yml:1]: ok

File: .github/workflows/publish-helm.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+on:
+  push:
+    tags:
+      - '*'
+    paths:
+      - '.github/workflows/publish-helm.yml'
```
Comment (aslafy-z) [inline .github/workflows/publish-helm.yml:7]: add `workflow_dispatch:` so we can manually deploy the chart (for testing purposes)

Comment (HamzaZo) [reply .github/workflows/publish-helm.yml:7]: done

**Comment by aslafy-z** [comment]
This PR could get some documentation in the README, before the configuration options.

**Comment by aslafy-z** [comment]
https://carrefour-group.github.io/azure-resourcemanager-exporter/index.yaml

File: README.md [hunk 5]
```diff
@@ -7,9 +7,34 @@ Azure ResourceManager Exporter
 
 Prometheus exporter for Azure Resources and information.
 
-Configuration
+Installation
 -------------
 
+[Helm 3](https://helm.sh/) must be installed to use the chart. Once Helm is set up properly, Add the repo as follows:
```
Comment (aslafy-z) [inline README.md]:
```suggestion
[Helm 3](https://helm.sh/) must be installed to use the chart. Once Helm is set up properly, add the repo as follows:
```

File: README.md [hunk 6]
```diff
@@ -7,9 +7,34 @@ Azure ResourceManager Exporter
...
+```shell
+$ helm uninstall [RELEASE_NAME] 
+```
+
+Exporter Configuration
+---------------------
+
 Normally no configuration is needed but can be customized using environment variables.
 
 | Environment variable              | DefaultValue                | Description                                                       |
```
Comment (aslafy-z) [inline README.md:40]: Can you add a `Contributing` section with a `Release` subsection that mention what needs to be bumped in order to release correctly?
helm `appVersion`, `version` and helm values `image.tag` should be bumped

File: README.md [hunk 7]
```diff
@@ -85,3 +110,15 @@ Metrics
 | `azurerm_iam_principal_info`                   | IAM               | Azure IAM Principal information                                                       |
 | `azurerm_graph_app_info`                       | Graph             | AzureAD graph application information                                                 |
 | `azurerm_graph_app_credential`                 | Graph             | AzureAD graph application credentials (create,expiry) information                     |
+
+
+Contributing
+------------
+We welcome any contributions from the community with open arms, If you're planning a new feature, please file an issue to discuss first.
```
Comment (aslafy-z) [inline README.md]:
```suggestion
We welcome any contributions from the community with open arms. If you're planning a new feature, please file an issue to discuss first.
```

File: README.md [hunk 8]
```diff
@@ -85,3 +110,15 @@ Metrics
...
+
+How to Release
+--------------
+To release a new version of the helm chart, you need to bump :
```
Comment (aslafy-z) [inline README.md]:
```suggestion
To release a new version of the helm chart, you need to bump:
```

File: README.md [hunk 9]
```diff
@@ -85,3 +110,15 @@ Metrics
...
+ * `appVersion` & `version` fields in `Chart.yaml` file.
```
Comment (aslafy-z) [inline README.md]:
```suggestion
 * `appVersion` and `version` fields in `Chart.yaml` file.
```

**Review by aslafy-z — APPROVED** [review]
/lgtm

**Pull request merged**